src/App.tsx

```diff
@@ -1,26 +1,308 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+/* eslint-disable @typescript-eslint/no-shadow */
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import { Todo } from './components/todo.component/todo.types';
+import { TodoComponent } from './components/todo.component/todo.component';
+import { TodoStatus } from './components/filter.component/filter.status';
+import { ErrorTypes } from './components/error.component/error';
+import * as Services from './api/todos';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorTitle, setErrorTitle] = useState<string | null>(null);
+  const [status, setStatus] = useState<TodoStatus>(TodoStatus.All);
+  const [newTodoTitle, setNewTodoTitle] = useState<string>('');
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [toggleAllClicked, setToggleAllClicked] = useState<boolean>(false);
+  const [isLoadingAll, setIsLoadingAll] = useState(false);
+
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    Services.getTodos()
+      .then(setTodos)
+      .catch(() => setErrorTitle(ErrorTypes.UnableToLoadTodos));
+  }, []);
+
+  useEffect(() => {
+    if (errorTitle !== null) {
+      setTimeout(() => setErrorTitle(null), 4000);
+    }
+  }, [errorTitle]);
+
+  useEffect(() => {
+    if (!toggleAllClicked && errorTitle !== ErrorTypes.UnableToUpdateTodo) {
+      inputRef.current?.focus();
+    } else {
+      setToggleAllClicked(false);
+    }
+  }, [todos, errorTitle, toggleAllClicked]);
+
+  const handleAddTodo = useCallback(
+    (event: React.FormEvent<HTMLFormElement>) => {
+      event.preventDefault();
+      const trimmedTitle = newTodoTitle.trim();
+
+      if (!trimmedTitle) {
+        setErrorTitle(ErrorTypes.TitleError);
+
+        return;
+      }
+
+      const newTempTodo: Todo = {
+        id: 0,
+        title: trimmedTitle,
+        completed: false,
+        userId: Services.USER_ID,
+      };
```

Comment:
[Optional fix] please modify your Todo creation logic, that this default Todo will not be defined inside the component